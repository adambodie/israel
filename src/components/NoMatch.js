import React from 'react';
import { Row, Column } from 'react-foundation';
import { mainLink } from '../data/articles';


const NoMatch = ({ location }) => {
	return (
		<div className='404'>
			<h1>Page Not Found</h1>
			<Row className='display grid-x noMatch'>
				<Column small={4}>  
					<img src={`${mainLink}/bathroom.jpg`} alt='bathroom' />
				</Column>
				<Column small={8}>
					<p>If you're looking for the bathroom, you&apos;ve come to the right place.</p>
					<p>If you're looking for the link for <strong>{location.pathname}</strong>, it doesn't exist.</p>
					<p>If you believe {location.pathname} exists, please contact the <a href='mailto:fake-email-address@xyz.com'>administrator</a>.</p>
					<p>Otherwise, go back and try again.</p>
				</Column>
			</Row>
		</div>
	);
};

export default NoMatch;
