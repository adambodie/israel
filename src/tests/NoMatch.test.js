import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import NoMatch from '../components/NoMatch';

it('renders without crashing', () => {
	const location = { pathname: '/dashboard/' };
	const div = document.createElement('div');
	ReactDOM.render(<NoMatch location={location} />, div);
	ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
	const location = { pathname: '/dashboard/' };
	const tree = renderer.create(<NoMatch location={location} />).toJSON();
	expect(tree).toMatchSnapshot();
});

