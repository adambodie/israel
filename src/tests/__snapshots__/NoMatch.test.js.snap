// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="404"
>
  <h1>
    Page Not Found
  </h1>
  <div
    className="row display grid-x noMatch"
  >
    <div
      className="columns small-4"
    >
      <img
        alt="bathroom"
        src="https://s3-us-west-2.amazonaws.com/birthright-israel.bodiewebdesign.com/images/main/bathroom.jpg"
      />
    </div>
    <div
      className="columns small-8"
    >
      <p>
        If you're looking for the bathroom, you've come to the right place.
      </p>
      <p>
        If you're looking for the link for 
        <strong>
          /dashboard/
        </strong>
        , it doesn't exist.
      </p>
      <p>
        If you believe 
        /dashboard/
         exists, please contact the 
        <a
          href="mailto:fake-email-address@xyz.com"
        >
          administrator
        </a>
        .
      </p>
      <p>
        Otherwise, go back and try again.
      </p>
    </div>
  </div>
</div>
`;
